### PR TITLE
Updates to add network security perimeter

### PR DESCRIPTION
**Pull request by caleteeter**
## Purpose

PR to add support for network security perimeter in learning mode for connected resources.

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
Ran deployment with default parameters with success.

* Test the code

N/A

## What to Check
Verify that the following are valid
N/A

### CODE REVIEW

File: bicep/novnet/main.bicep [hunk 1]
```diff
@@ -179,6 +182,9 @@ param storageAccountName string = ''
 @description('Specifies the access tier of the Azure Storage Account resource. The default value is Hot.')
 param storageAccountAccessTier string = 'Hot'
 
+@description('Specifies whether the Azure Storage Account resource allows public access. The default value is enabled.')
+param storageAccountAllowPublicAccess string = 'Enabled'
```
Comment (caleteeter) [inline bicep/novnet/main.bicep:186]: @paolosalvatori, wanted to get your thoughts on this. In this PR the default would be to have public access using network security perimeter in learning mode.  Do you think this is a good default, if we disable by default, then users will need to add to the parameters to override?

Comment (paolosalvatori) [reply bicep/novnet/main.bicep:186]: I would keep the ordinary default value storageAccountAllowPublicAccess  = 'Enabled', and set its value to Disabled via parameter. I would also drive the creation / enablement of the network security perimeter  via parameter. Thoughts?

**Comment by paolosalvatori** [comment]
@caleteeter please let me know when both PRs are ready to be reviewed and approved or if you still have some changes to do. Thanks 🙂

File: bicep/novnet/main.bicep (hunk 1 above)
Comment (caleteeter) [reply bicep/novnet/main.bicep:186]: @adrianmo @paolosalvatori resolved this.

**Review by paolosalvatori — APPROVED** [review]

**Pull request merged**